service/src/main/java/com/bankbazaar/kafka/service/config/KafkaStreamsConfig.java

```diff
@@ -0,0 +1,151 @@
+                                    if (file.exists()) {
+                                        fileData.setStatus(Status.FAILURE);
+                                        fileStatusService.update(fileData);
+                                        return false;
+                                    }
+                                    return true;
+                                }
+                                catch (Exception exception)
+                                {
+                                    log.error("retrying",exception);
+                                    throw exception;
+                                }
+                                },
+
+                            context -> {
+                                log.error("retries exhausted");
+                                FileStatusEntity fileData = new FileStatusEntity();
+                                fileData.setId(value.getId());
+                                fileData.setStatus(Status.ERROR);
+                                fileStatusService.update(fileData);
+                                return false;
+                            }
+                            );
+                }
+                );
+    }
+
+    /**
+     * Consumes data from File_Processor topic.
+     * Create.csv file and populate file using data.
+     * Set status to SUCCESS if file IS created successfully.
+     * Set status to ERROR if exception occurs during file creation.
+     * Retry a maximum of 1 time if exception occurs;
+     */
+    @Bean
+    public Consumer<KStream<String,Data>> consumer(@Qualifier("streamRetryTemplate") RetryTemplate retryTemplate)
+    {
+        return kStream -> kStream.foreach((key, value) ->
+                {
+                    try {
+                        retryTemplate.execute(
+                                retryContext -> {
+                                    try {
+                                        ClassLoader classLoader = getClass().getClassLoader();
+                                        FileStatusEntity fileData = new FileStatusEntity();
+                                        fileData.setId(value.getId());
+                                        File file = new File(classLoader.getResource(".").getFile() + value.getFileName());
+                                        FileWriter fileWriter = new FileWriter(file,false);
+                                        CSVWriter CsvWriter = new CSVWriter(fileWriter);
+                                        CsvWriter.writeNext(value.getHeaders());
+                                        for (String[] element : value.getData()) {
+                                            CsvWriter.writeNext(element);
+                                        }
+                                        CsvWriter.close();
+                                        fileData.setStatus(Status.SUCCESS);
+                                        fileStatusService.update(fileData);
```

Comment:
Post the initial creation we are only updating the status. I think it will be better to have a method updateStatus(Long id, Status status) for this purpose

Reply:
this has been implemented